src/functions/reportGen.ts

```diff
@@ -16,22 +23,38 @@ const reportGen: Handler = async (event: any, context?: Context, callback?: Call
         console.error("ERROR: event is not defined.");
         return;
     }
-
     const s3BucketService: S3BucketService = Injector.resolve<S3BucketService>(S3BucketService);
     const reportService: ReportGenerationService = Injector.resolve<ReportGenerationService>(ReportGenerationService);
     const retroUploadPromises: Array<Promise<ManagedUpload.SendData>> = [];
+    const testStationsService: TestStationsService = Injector.resolve<TestStationsService>(TestStationsService);
+    const notifyClient = new NotifyClient(Configuration.getInstance().getGovNotifyConfig().api_key);
+    const notifyService: NotificationService = new NotificationService(notifyClient);
+    const notificationData: NotificationData = new NotificationData();
+    let testResultsList: any;
 
     event.Records.forEach((record: any) => {
         const visit: any = JSON.parse(record.body);
         const retroUploadPromise = reportService.generateATFReport(visit)
-        .then((generationServiceResponse: { fileName: string, fileBuffer: Buffer}) => {
-
+        .then((generationServiceResponse: { fileName: string, fileBuffer: Buffer, testResults: any}) => {
+            testResultsList = generationServiceResponse.testResults;
             return s3BucketService.upload(`cvs-atf-reports-${process.env.BUCKET}`, generationServiceResponse.fileName, generationServiceResponse.fileBuffer)
             .then((result: any) => {
-                return result;
+               return testStationsService.getTestStationEmail(visit.testStationPNumber)
+                   .then((response: any) => {
+                       const sendNotificationData = notificationData.generateActivityDetails(visit, testResultsList);
+                       return notifyService.sendNotification(sendNotificationData, response[0].testStationEmails).then(() => {
+                           return result;
+                       }).catch((error: any) => {
+                           console.log(error);
```

Comment:
These log the error, but don't actual return a failure response of any kind back up the chain. Is that our preferred outcome?

Reply:
I think if an error occurs then the event should be put back in the queue in order to retry submitting it, however I think there should be a maximum number of attempts and then an error to be captured in the logs. @akothiyal-deloitte what do you think?

Reply:
The repopulation of the queue was fixed by throwing an error if the report can't be generated.
Because we don't have the technical design for what errors should we throw, we are just logging them

Reply:
@akothiyal-deloitte thoughts?

Reply:
Dont believe this is correct, why are we swallowing an error? if something goes wrong we need throw back relevant error/exception, which in this specific case should result in message appearing back on queue. 
We cant fail silently like this.
@acozianu  @adpetrescu @KHodderDVSA

Reply:
@akothiyal-deloitte That's what I was thinking as well. What error should @acozianu throw back?